### PR TITLE
Language switcher

### PR DESCRIPTION
**Pull request by josue2591**
# Ticket(s)

- [FP-243: language switcher](https://fourkitchens.clickup.com/t/36718269/FP-243)

## Purpose

Create a language switcher component

### Functional Testing

- [x] Go to http://localhost:6006/?path=/story/elements-languageswitcher--default
- [ ] The language switcher should work as expected
  - [ ] Should open only with click
  - [ ] The active language switcher should be at the begining of the list

## Video: https://www.loom.com/share/b4de292982174734b9ab86898dc8ba38?sid=1fe9cd5d-e055-424a-81df-c064b8d998f1

### CODE REVIEW

**Comment by fkbender** [comment]
Task linked: [FP-243 Header Language Switcher](https://app.clickup.com/t/36718269/FP-243)

File: README.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ export default {
 - [Link](./docs/link.md)
 - [Button](./docs/button.md)
 - [Typography](./docs/typography.md)
+- [Typography](./docs/typography.md)
```
Comment (josue2591) [inline README.md:36]:
```suggestion
```

File: src/lib/components/02-components/menus/language-switcher/language-switcher.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  const [items, setItems] = useState<MenuItemProps[]>([]);
+  useEffect(() => {
+    let activeLocaleText = matchLanguage[activeLocale];
+    if (activeLocaleText.length > 10) {
+      activeLocaleText = `${activeLocaleText.slice(0, 10)}...`;
+    }
+
+    const belowItems: MenuItemProps = { title: activeLocaleText, below: [] };
+
+    locales.sort((a, b) => {
```
Comment (cienvaras) [inline src/lib/components/02-components/menus/language-switcher/language-switcher.tsx:53]: I don't think this is necessary. You can use `findIndex()` to find the index of the active locale and `splice()` to remove the item. Then, you can add the item manually at the beginning of the array using `unshift()`.

File: src/lib/components/02-components/menus/language-switcher/language-switcher.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  'zh-hant': '繁體中文',
+  ar: 'العربية',
+};
+
+export default function LanguageSwitcher({
+  activeLocale,
+  locales,
+  url,
+}: LanguageSwitcherProps) {
+  const [items, setItems] = useState<MenuItemProps[]>([]);
```
Comment (cienvaras) [inline src/lib/components/02-components/menus/language-switcher/language-switcher.tsx:44]: I'm pretty sure you don't need a state for this. Just pass the item list you process below directly to the `SecondaryMenu` component

File: src/lib/components/02-components/menus/language-switcher/language-switcher.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    const belowItems: MenuItemProps = { title: activeLocaleText, below: [] };
+
+    locales.sort((a, b) => {
+      if (a === activeLocale) return -1;
+      if (b === activeLocale) return 1;
+      return 0;
+    });
+
+    locales.forEach((locale) => {
```
Comment (cienvaras) [inline src/lib/components/02-components/menus/language-switcher/language-switcher.tsx:59]: Use a map for this to generate the array and then after that build the `bellowItems` object.

File: src/lib/components/02-components/menus/language-switcher/language-switcher.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        active: activeLocale === locale,
+      });
+    });
+    setItems([belowItems]);
+  }, [activeLocale]);
+
+  return (
+    <SecondaryMenu
+      items={items}
+      menuClass="language-switcher"
```
Comment (cienvaras) [inline src/lib/components/02-components/menus/language-switcher/language-switcher.tsx:72]: What's this class for?

**Review by cienvaras — CHANGES_REQUESTED** [review]
Looks pretty good, just check the inline comments for some things that can be improved.

File: src/lib/components/02-components/menus/language-switcher/language-switcher.tsx (hunk 2 above)
Comment (josue2591) [reply src/lib/components/02-components/menus/language-switcher/language-switcher.tsx:53]: Done

File: src/lib/components/02-components/menus/language-switcher/language-switcher.tsx (hunk 4 above)
Comment (josue2591) [reply src/lib/components/02-components/menus/language-switcher/language-switcher.tsx:59]: Done

File: src/lib/components/02-components/menus/language-switcher/language-switcher.tsx (hunk 5 above)
Comment (josue2591) [reply src/lib/components/02-components/menus/language-switcher/language-switcher.tsx:72]: Is used in the `secondary-menu.tsx` component is used there to create keys and id's

File: src/lib/components/02-components/menus/language-switcher/language-switcher.tsx (hunk 3 above)
Comment (josue2591) [reply src/lib/components/02-components/menus/language-switcher/language-switcher.tsx:44]: Done

File: src/lib/components/02-components/menus/language-switcher/language-switcher.tsx (hunk 4 above)
Comment (josue2591) [reply src/lib/components/02-components/menus/language-switcher/language-switcher.tsx:59]: Done

**Review by mcortes19 — APPROVED** [review]

**Pull request merged**